src/index.ts

```diff
@@ -153,6 +149,18 @@ function initConfig(userConfig: Forceable<Config>): TopLevelConfig {
           process.env.GAE_MINOR_VERSION
         ),
       },
+      /**
+       * Our TypeScript interface suggests that only credentials, keyFilename,
+       * and projectId are accepted, but by passing the entire object to the
+       * Trace Writer, we can allow users to supply other fields that are
+       * publicly supported by the Google Auth Library.
+       */
+      authOptions: Object.assign({}, mergedConfig, {
+        projectId: lastOf<string | undefined>(
+          mergedConfig.projectId,
+          process.env.GCLOUD_PROJECT
```

Comment:
Doesn't the auth library already consult this environment variable?

Reply:
Yeah, we do.

Reply:
It's unfortunate but I realized that Trace Agent has historically regarded `GCLOUD_PROJECT` as taking precedence over the `projectId` config option, while the auth library does the [opposite](https://github.com/googleapis/google-auth-library-nodejs/blob/197dd87f0019ac4314043e961eb8b3ba78670c4c/src/auth/googleauth.ts#L171). Maybe it's possible that we change this now, though.

Reply:
That would be semver major :/. Not worth it.